src/Lykke.Service.MarketMakerReports.AzureRepositories/AuditMessageRepository.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using AutoMapper;
+using AzureStorage;
+using Lykke.Service.MarketMakerReports.Core.Domain.AuditMessages;
+using Lykke.Service.MarketMakerReports.Core.Repositories;
+using Microsoft.WindowsAzure.Storage.Table;
+
+namespace Lykke.Service.MarketMakerReports.AzureRepositories
+{
+    public class AuditMessageRepository : IAuditMessageRepository
+    {
+        private readonly INoSQLTableStorage<AuditMessageEntity> _storage;
+
+        public AuditMessageRepository(INoSQLTableStorage<AuditMessageEntity> storage)
+        {
+            _storage = storage;
+        }
+
+        public async Task<IReadOnlyList<AuditMessage>> GetAsync(DateTime? startDate, DateTime? endDate, string clientId = null)
+        {
+            string filter = null;
+            
+            if (startDate.HasValue && endDate.HasValue)
```

Comment:
This code can be simplified. Create collection of filters and combine them in one place.

Reply:
Sure, did the simplification.